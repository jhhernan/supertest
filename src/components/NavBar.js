import React from 'react';
import { mdiCart } from '@mdi/js';
import Icon from '@mdi/react';

function NavBar(props) {
    return (
        <div className="navbar">
            <div><a className="logo" href="/">superf√ºds</a></div>
            <input className="searchBox" type="text" placeholder="Busca marcas y productos" href="#news" />
            <div className="user">
                <div className="userCart" onClick={props.showCart}><Icon path={mdiCart} size={1} /></div>
                <span data-testid="user-name">Jairo Hernandez</span>
                <img className="profilePic" alt="avatar" src="avatar.png"/>
            </div>
        </div>
    )
}

export default NavBar;