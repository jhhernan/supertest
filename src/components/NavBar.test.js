import React from 'react';
import { render, cleanup, fireEvent, waitForElement } from '@testing-library/react';
import NavBar from './NavBar';


describe('NavBar component test', () => {

    it('renders the component', () => {
        const container = render(<NavBar  />)
        expect(container.firstChild).toMatchSnapshot()
    })

    it('display correctly ', () => {
        const { getAllByTestId, getByTestId } = render(
              <NavBar />
          );
        
        expect(getByTestId('user-name').textContent).toBe("Jairo Hernandez");        
    })
  });

