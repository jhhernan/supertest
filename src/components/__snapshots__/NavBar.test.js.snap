// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component test renders the component 1`] = `undefined`;
